common/txmgr/address_state.go

```diff
@@ -350,6 +350,8 @@ func (as *addressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) moveTx
 		txAttempt.BroadcastBeforeBlockNum = &blockNum
 	}
 	tx.State = TxConfirmed
+	as.confirmedTxs[tx.ID] = tx
```

Comment:
Just want to confirm here too that tx are only moved from Unconfirmed to Confirmed?